pkg/apis/maistra/conversion/extension_providers_test.go

```diff
@@ -176,3 +177,38 @@ meshConfig:
 		},
 	}
 }
+
+func TestInvalidValueType(t *testing.T) {
+	helmValues := v1.HelmValues{}
+	if err := helmValues.UnmarshalYAML([]byte(`
+meshConfig:
+  extensionProviders:
+  - envoyExtAuthzHttp:
+      service: "test"
+      port: "80"
+`)); err != nil {
+		t.Fatalf("failed to parse helm values: %s", err)
+	}
+
+	specV1 := v1.ControlPlaneSpec{
+		Version: "v2.4",
+		Istio:   &helmValues,
+	}
+	var specV2 v2.ControlPlaneSpec
+	err := Convert_v1_ControlPlaneSpec_To_v2_ControlPlaneSpec(&specV1, &specV2, nil)
+	if err != nil {
```

Comment:
nit: could we consistently initialize errors in `if` conditions, like in the previous one?

Reply:
I'm not a fan of using the `if err := doSth(); err != nil {` as it makes the code harder to scan (it hides the action in the middle of boilerplate code), so I don't think we should mandate the use of this pattern. But I changed this nevertheless.

Reply:
On the other hand, this approach enforces to check error and makes sure it's not overwritten then - what may cause not checking the first error. Anyway, I didn't want to mandate this approach in general. It's just a bit distracting when I see different conventions in the scope of a single function.